internal/pkg/consensus/validation.go

```diff
@@ -47,54 +47,41 @@ func init() {
 	errNonceTooHighCt = metrics.NewInt64Counter("consensus/msg_nonce_high_err", "Number of messages with nonce too high")
 }
 
-// DefaultMessageValidator validates incoming signed messages.
-type DefaultMessageValidator struct {
-	allowHighNonce bool
+// MessageSelectionChecker checks for miner penalties on signed messages
+type MessagePenaltyChecker struct {
```

Comment:
Mismatched name/comment.

I would rename to something more positive like message qualifier or something, not specifically tied to the penalties we know happen in the VM. It's role is to select messages that make sense to include in a block, with reference to the parent state.

Reply:
I would argue that this does not belong in the consensus package, but closer to block production. It needs access to state, but has nothing else to do with consensus.

Reply:
This is wrapping the consensus rules which the mining messagequalifier uses but is not restricted to.  In the future I imagine the qualifier will own an instance of this type instead of being this type.  I am under the impression we are going to need to use these rules somewhere during validation (i.e. consensus) not just in mining message selection in order to actually extract the penalty.  

Odds are good I'm getting something wrong here but I'm going to move to leave things more obviously wrong now and fix them once I'm back up to speed.